src/linter/rules/seeded-randomness.ts

```diff
@@ -0,0 +1,140 @@
+import type { LintingResult, LintingRule } from '../linter-format';
+import { LintingPrettyPrintContext , LintingCertainty } from '../linter-format';
+import type { SourceRange } from '../../util/range';
+import type { MergeableRecord } from '../../util/objects';
+import { Q } from '../../search/flowr-search-builder';
+import { formatRange } from '../../util/mermaid/dfg';
+import { Enrichment, enrichmentContent } from '../../search/search-executor/search-enrichers';
+import type { Identifier } from '../../dataflow/environments/identifier';
+import { FlowrFilter } from '../../search/flowr-search-filters';
+import { DefaultBuiltinConfig } from '../../dataflow/environments/default-builtin-config';
+import type { DataflowGraphVertexFunctionCall } from '../../dataflow/graph/vertex';
+import { getReferenceOfArgument } from '../../dataflow/graph/graph';
+import { CascadeAction } from '../../queries/catalog/call-context-query/cascade-action';
+import { recoverName } from '../../r-bridge/lang-4.x/ast/model/processing/node-id';
+import { LintingRuleTag } from '../linter-tags';
+import type { BuiltInFunctionDefinition } from '../../dataflow/environments/built-in-config';
+import { getValueOfArgument } from '../../queries/catalog/call-context-query/identify-link-to-last-call-relation';
+import { RType } from '../../r-bridge/lang-4.x/ast/model/type';
+
+export interface SeededRandomnessResult extends LintingResult {
+	function: string
+    range: SourceRange
+}
+
+export interface SeededRandomnessConfig extends MergeableRecord {
+	/**
+	 * A set of functions and variables whose invocation or assignment causes a random seeded to be set.
+	 * Each entry has a `type`, which is either `function` or `assignment`, and a `name`, which is the name of the function or variable.
+	 * The default value for this is the function `set.seed` and the variable `.Random.seed`.
+	 */
+    randomnessProducers: {type: 'function' | 'assignment', name: string }[]
+	/**
+	 * A set of randomness consumer function names that require a seed to be set prior to invocation.
+	 */
+    randomnessConsumers: string[]
+}
+
+export interface SeededRandomnessMeta extends MergeableRecord {
+	consumerCalls:                 number
+	callsWithFunctionProducers:    number
+	callsWithAssignmentProducers:  number
+	callsWithNonConstantProducers: number
+}
+
+const assignments = DefaultBuiltinConfig.filter(b => b.type === 'function' && b.processor == 'builtin:assignment') as BuiltInFunctionDefinition<'builtin:assignment'>[];
```

Comment:
This should not happen at loading time given that we can reconfigure the BuiltIn config